### PR TITLE
CEDS-1695 Fix for Pointer Numerical Field Sections

### PR DESCRIPTION
**Pull request by ed0906**
We used to presume a numerical section meant it was a sequence.
This assumption is not correct.

Sequence sections are now rendered in JSON as `#1` (for index 1)
e.g.
Pointer Before: `abc.1.def`
Pointer After: `abc.#1.def`

It means we can convert to and from the string/json version and retain the "section type"

### CODE REVIEW

File: app/models/Pointer.scala [hunk 1]
```diff
@@ -16,25 +16,53 @@
...
+  // Converts a pointer to it's string form preserving the type
+  // e.g. ABC.DEF.#1.GHI (if the pointer contains a sequence with index 1)
+  // e.g. ABC.DEF.GHI (if the pointer doesnt contain a sequence)
+  override def toString: String = sections.map(_.toString).mkString(".")
 }
 
 object Pointer {
   implicit val format: Format[Pointer] =
-    Format(Reads(js => js.validate[String].map(string => Pointer(string))), Writes(pointer => JsString(pointer.value)))
+    Format(Reads(js => js.validate[JsString].map(string => Pointer(string.value))), Writes(pointer => JsString(pointer.toString)))
```
[the remark this answers is not in the document]
Comment (PatrykRudnicki) [reply app/models/Pointer.scala]: ahh, it's the same like previous implementation 🤦‍♂ Friday...

[the remark this answers is not in the document]
Comment (ed0906) [reply app/models/Pointer.scala]: No, you're right it should just be String

[the remark this answers is not in the document]
Comment (ed0906) [reply app/models/Pointer.scala]: fixed

[the remark this answers is not in the document]
Comment (ed0906) [reply app/models/Pointer.scala]: Fixed in the back end too

**Pull request merged**